coordinator/src/node/unrealized_pnl.rs

```diff
@@ -0,0 +1,68 @@
+use crate::db;
+use crate::node::Node;
+use crate::position::models::Position;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::PooledConnection;
+use diesel::PgConnection;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use trade::bitmex_client::BitmexClient;
+use trade::bitmex_client::Quote;
+use trade::cfd::calculate_pnl;
+use trade::Direction;
+
+pub async fn sync(node: Node) -> Result<()> {
+    let mut conn = node.pool.get()?;
+
+    let positions = db::positions::Position::get_all_open_or_closing_positions(&mut conn)?;
+    let current_quote = BitmexClient::get_quote(&OffsetDateTime::now_utc())
```

Comment:
🔧 We shouldn't fetch new quotes for a position that is state Closing. The execution price should already be set.

Reply:
well... yes and no. The execution price is set, but we only save the *actual* pnl that was collected when we have a `ClosedContract` (based on said contract).
There is no guarantee that a position is `Closing` will actually be closed with that execution price, so I'm not sure it's worth jumping through hoops of setting the execution price and using that. 

I can do it if we really feel it's necessary :)

Reply:
> There is no guarantee that a position in Closing will actually be closed with that execution price.

Not sure I am following. 

Are you referring to a technical reason why we can't guarantee the the execution price?

A match has already been found so there should be no other reason on why we couldn't guarantee the price. Am I missing something?

Reply:
> Are you referring to a technical reason why we can't guarantee the the execution price?

There are multiple reasons why it could fail, I am not sure you can classify all of them as technical. A scenario where e.g. a lot of trades happen at the same time with the same maker could lead liquidity problems with the maker which could lead to execution failure. Same for liquidity on the coordinator.

> A match has already been found so there should be no other reason on why we couldn't guarantee the price. Am I missing something?

Example: We found a match, the app contacts the coordinator but the maker is down / cannot fulfill the trade. The execution just can't get guaranteed. There are other (technical) reasons why it could fail, e.g. the app being offline for some time because the user lost network. 

However, after some thought I think saving the execution price is the right way to go because we will need it for sanity/auditing/validation. We have to make sure that we also remove it in case execution failed. (not part of this PR)
I will see to add saving the execution price with the position once known and will add a separate code path for that reports the `unrealized_pnl` based on the `execution_price` if the `execution_price` is known.

Reply:
Might be good to read https://github.com/get10101/10101/pull/966#issuecomment-1644850594 as well :)